internal/service/advancedcluster/resource_advanced_cluster_test.go

```diff
@@ -1106,10 +1100,16 @@ func configReplicaSetAWSProvider(projectID, name string, diskSizeGB, nodeCountEl
 func checkReplicaSetAWSProvider(projectID, name string, diskSizeGB, nodeCountElectable int, checkDiskSizeGBInnerLevel, checkExternalID bool) resource.TestCheckFunc {
 	additionalChecks := []resource.TestCheckFunc{
 		resource.TestCheckResourceAttr(resourceName, "retain_backups_enabled", "true"),
-		resource.TestCheckResourceAttrWith(resourceName, "replication_specs.0.region_configs.0.electable_specs.0.disk_iops", acc.IntGreatThan(0)),
+	}
+	if config.AdvancedClusterV2Schema() {
+		// Manual test fix to remove electable_specs.0 for TPF
+		additionalChecks = append(additionalChecks,
+			resource.TestCheckResourceAttrWith(resourceName, "replication_specs.0.region_configs.0.electable_specs.disk_iops", acc.IntGreatThan(0)),
+		)
 	}
 	if !config.AdvancedClusterV2Schema() { // TODO: data sources not implemented for TPF yet
 		additionalChecks = append(additionalChecks,
+			resource.TestCheckResourceAttrWith(resourceName, "replication_specs.0.region_configs.0.electable_specs.0.disk_iops", acc.IntGreatThan(0)),
```

Comment:
@EspenAlbert  most of the checks like:
```
"replication_specs.0.region_configs.0.electable_specs.0.disk_size_gb": fmt.Sprintf("%d", diskSizeGB)
```

can go directly in the map instead of additional params so we don't need to call attrNameToSchemaV2 individually.

only for "special" checks like TestCheckResourceAttrWith, this can't be done